src/main.rs

```diff
@@ -784,19 +797,8 @@ fn get_bibtex_entry_from_record(
 }
 
 /// Get keys equivalent to a given key that are valid BibTeX citation keys.
-fn get_valid_keys(
-    record_db: &mut RecordDatabase,
-    key: &str,
-) -> Result<Vec<String>, rusqlite::Error> {
-    let entry = record_db.initialize_row(&RecordId::from(key));
-    if let Ok(RecordsTableRow::Exists(row)) = entry {
-        let alternative_keys = row
-            .apply(row::get_referencing_keys)?
-            .into_iter()
-            .filter(|k| is_entry_key(k))
-            .collect();
-        row.commit()?;
-        return Ok(alternative_keys);
-    }
-    Ok(vec![])
+fn get_valid_keys(row: &RecordRow) -> Result<Vec<String>, rusqlite::Error> {
+    let mut alternative_keys = row.apply(row::get_referencing_keys)?;
```

Comment:
Maybe `referencing_keys` instead of `alternative_keys`? Also maybe a better function name is `get_valid_referencing_keys`?